### ISSUE TITLE
feat: option to pass runnerFlags like '--disableupdate' to runner

### ISSUE THREAD

**Issue by beckjim** [CloudSnorkel/cdk-github-runners#266]
Why:
For ephemeral runners, it often makes sense to control runner updates by yourself.
So passing '--disableupdate' to the runnersetup would a helpful option.
See [https://docs.github.com/en/actions/hosting-your-own-runners/autoscaling-with-self-hosted-runners#controlling-runner-software-updates-on-self-hosted-runners](url) for details.

Thx!

br

**Comment by kichik** [comment]
The runner automatically uses `--disableupdate` when `runnerVersion` is specified and is not set to `RunnerVersion.latest()`. For example:

```
const myBuilder = new CodeBuildImageBuilder(this, 'image builder', {
  dockerfilePath: FargateRunner.LINUX_X64_DOCKERFILE_PATH,
  runnerVersion: RunnerVersion.specific('2.291.0'),
  rebuildInterval: Duration.days(14),    
});

const myProvider = new FargateRunnerProvider(this, 'fargate runner', {
  label: 'customized-fargate',
  imageBuilder: myBuilder,
});

// create the runner infrastructure
new GitHubRunners(this, 'runners', {
  providers: [myProvider],
});
```

Either way the image is rebuilt every `rebuildInterval`. Unless you specify a runner version, the image should always have a reasonably up-to-date version. You can disable this as well by setting `rebuildInterval` to `Duration.days(0)`.

Also see https://github.com/CloudSnorkel/cdk-github-runners/issues/112#issuecomment-1272663673

**Comment by beckjim** [comment]
Thanks for the quick reply. 

For our fargate runners, this was not working:
The runner version was set to the "shipped" (2.293.1) runner version of our GHES 
After updating GHES, our fargate runners wanted to update themselves ...

I'm not js/ts dev, but I tried to check the code for fargate runners.
For me it looks like the implemented logic is working the opposite way. Maybe I'm wrong.
[https://github.com/CloudSnorkel/cdk-github-runners/blob/main/src/providers/fargate.ts#L450](url)

```typescript
  private runCommand(): string[] {
    let runnerFlags = '';
    if (this.image.runnerVersion.is(RunnerVersion.latest())) {
      runnerFlags = '--disableupdate';
    }
 ```
 
 thx!

**Comment by kichik** [comment]
You're right. That is backwards. It adds the `--disableupdate` flag when the runner version is set to latest. It should add it when it's pinned to something that's not latest.
